lib/node_modules/@stdlib/string/base/reverse-grapheme-clusters/test/test.js

```diff
@@ -0,0 +1,77 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var reverse = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof reverse, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns an empty string if provided an empty string', function test( t ) {
+	t.strictEqual( reverse( '' ), '', 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function reverses the grapheme clusters of a provided string (ascii)', function test( t ) {
+	var out;
+
+	out = reverse( 'hello world' );
+	t.strictEqual( out, 'dlrow olleh', 'returns expected value' );
+
+	out = reverse( '!!!*' );
+	t.strictEqual( out, '*!!!', 'returns expected value' );
+
+	out = reverse( 'Hello World' );
+	t.strictEqual( out, 'dlroW olleH', 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function reverses the grapheme clusters of a provided string (Unicode)', function test( t ) {
+	var out;
+
+	out = reverse( 'अनुच्छेद', 1 );
+	t.strictEqual( out, 'दछेच्नुअ', 'returns expected value' );
```

Comment:
I think this answers my question above for `reverse-code-points`. This string is comprised of grapheme clusters, not just surrogate pairs.